### PR TITLE
Sync legends of LayoutPlot

### LINKED ISSUE #5674: Support syncing between visibility legend labels
As shown in https://github.com/holoviz/hvplot/issues/937#issuecomment-1498749914 it is possible to sync the visibility between lines in different figures. 

This seems like a great addition to have in Holoviews. 

https://user-images.githubusercontent.com/19758978/230332683-

### PR DESCRIPTION
**Pull request by Hoxbro**
Closes #5674 

TODO: 
- [x] Need to add some tests
- [x] Work if click_policy is `hide` (replace `muted` with `visibility`). 
- [x] Check whether `sync_location` is needed other places
- [x] Work with column, row

``` python
import holoviews as hv
import hvplot.pandas
import numpy as np
import pandas as pd
from holoviews.plotting.bokeh.util import select_legends

df = pd.DataFrame(np.arange(10)[:, None] * np.array([1, 2, 3]), columns=list("ABC"))
plot = (df.hvplot() + df.hvplot() + df.hvplot() + df.hvplot()).cols(2)
select_legends(plot, 0, "top_left")
```

https://github.com/holoviz/holoviews/assets/19758978/91817cc3-27ed-45dd-9e3b-27df7ed09134

### CODE REVIEW

File: holoviews/plotting/bokeh/element.py [hunk 1]
```diff
@@ -1170,6 +1170,8 @@ def _init_glyph(self, plot, mapping, properties):
             plot_method = plot_method[int(self.invert_axes)]
         if 'legend_field' in properties and 'legend_label' in properties:
             del properties['legend_label']
+        if "name" not in properties:
+            properties["name"] = properties.get("legend_label") or properties.get("legend_field", None)
```
Comment (ahuang11) [inline holoviews/plotting/bokeh/element.py:1174]: I think it defaults to None anyways.
```suggestion
            properties["name"] = properties.get("legend_label") or properties.get("legend_field")
```

File: holoviews/plotting/bokeh/util.py [hunk 2]
```diff
@@ -407,6 +410,79 @@ def merge(tool, group):
...
+    policy = mapping.get(next(iter(click_policies)))
+    code = f"dst.{policy} = src.{policy}"
+    for item in items.values():
+        for src, dst in permutations(item, 2):
+            src.js_on_change(
+                policy,
+                CustomJS(code=code, args=dict(src=src, dst=dst)),
+            )
+
+def one_legend(plot_layout, legend_no=0, legend_position="top_right"):
```
Comment (ahuang11) [inline holoviews/plotting/bokeh/util.py:459]: What if we have legend_no as a list so users can have it displayed on multiple figures and maybe rename function as `limit_legends`? Also thoughts on renaming to `figure_index`?

**Review by ahuang11 — COMMENTED** [review]
This will be useful! Just left some comments

File: holoviews/plotting/bokeh/element.py (hunk 1 above)
Comment (Hoxbro) [reply holoviews/plotting/bokeh/element.py:1174]: Correct. It is weird as I'm doing the right thing on the same line... It must have been some refactoring I did :man_shrugging:

File: holoviews/plotting/bokeh/util.py (hunk 2 above)
Comment (Hoxbro) [reply holoviews/plotting/bokeh/util.py:459]: I'm not opposed to that idea. The function name would definitely need a rename. Not sure if `limit_legends` is the right one, though.

**Review by philippjfr — APPROVED** [review]
Looks great!

**Comment by philippjfr** [comment]
One test at least would be nice though.

**Comment by Hoxbro** [comment]
I could "easily" add a test to `select_legends`, but it should mostly be seen as a small utility. 

I have a hard time figuring out to make a unit test for `sync_legends`, as the changes are made when clicking on the canvas with changes happening in the browser.

**Comment by philippjfr** [comment]
The unit test would just check that for different types of plots the CustomJS callback is set up correctly.

**Comment by Hoxbro** [comment]
That should also be easy :)

**Comment by dcherian** [comment]
Amazing thank you!

**Pull request merged**